Assets/PlayTests/NuGetPlayTests.cs

```diff
@@ -0,0 +1,76 @@
+using System;
+using System.Collections;
+using System.Runtime.InteropServices;
+using NugetForUnity;
+using NUnit.Framework;
+using UnityEngine;
+using UnityEngine.TestTools;
+
+[assembly: PrebuildSetup(typeof(NuGetPlayTests))]
+[assembly: PostBuildCleanup(typeof(NuGetPlayTests))]
+
+/// <summary>
+/// Play mode tests allow us to install NuGet packages with Native code before play mode starts, then when play mode
+/// runs the native libraries are available for use.
+///
+/// There seems to be some Unity internals that prevents an edit mode test from adding a Native library and finding it
+/// in the same run. 
+/// </summary>
+public class NuGetPlayTests : IPrebuildSetup, IPostBuildCleanup
+{
+    readonly NugetPackageIdentifier sqlite = new NugetPackageIdentifier("SQLitePCLRaw.lib.e_sqlite3", "2.0.7");
+
+    /// <summary>
+    /// This is the version number of sqlite with periods replaced with zeros.
+    ///
+    /// Note: Version of the SQLite library does not match the NuGet package version 
+    /// </summary>
+    private readonly int _expectedVersion = 3035005;
+
+    [UnityTest]
+    public IEnumerator InstallAndRunSqlite()
+    {
+        yield return new WaitForFixedUpdate();
+
+        // Test the actual library by calling a "extern" method
+        var version = sqlite3_libversion_number();
+        Assert.That(version, Is.EqualTo(_expectedVersion));
+    }
+
+    /// <summary>
+    /// Call to the SQLite native file, this actually tests loading and access the library when called.
+    ///
+    /// On windows the call to sqlite3_libversion returns a garbled string so use the integer
+    /// </summary>
+    /// <returns></returns>
+    [DllImport("e_sqlite3")]
+    private static extern int sqlite3_libversion_number();
+
+    public void Setup()
+    {
+        try
+        {
+            sqlite3_libversion_number();
+            Assert.Fail("e_sqlite3 dll loaded, but should not be available");
+        }
+        catch (DllNotFoundException)
+        {
+        }
+
+        // For windows we end up importing two identical DLLs temporarily, this causes an error log that NUnit detects
+        // and would fail the test if we don't tell it to ignore the Failing messages
+        LogAssert.ignoreFailingMessages = true;
+        NugetHelper.InstallIdentifier(sqlite);
```

Comment:
I hit an exception here when running this from Unity because somehow the NugetConfigFile is not loaded, was able to fix by running `NuGetHelper.LoadNuGetConfigFile` before this call to `InstallIdentifier`.

```
NullReferenceException: Object reference not set to an instance of an object
NugetForUnity.NugetHelper.GetCachedPackage (NugetForUnity.NugetPackageIdentifier packageId) (at Assets/NuGet/Editor/NugetHelper.cs:1166)
NugetForUnity.NugetHelper.GetSpecificPackage (NugetForUnity.NugetPackageIdentifier packageId) (at Assets/NuGet/Editor/NugetHelper.cs:1112)
NugetForUnity.NugetHelper.InstallIdentifier (NugetForUnity.NugetPackageIdentifier package, System.Boolean refreshAssets) (at Assets/NuGet/Editor/NugetHelper.cs:1255)
NuGetPlayTests.Setup () (at Assets/PlayTests/NuGetPlayTests.cs:63)
UnityEditor.TestTools.TestRunner.TestLauncherBase+<>c.<ExecutePreBuildSetupMethods>b__3_0 (UnityEngine.TestTools.IPrebuildSetup targetClass) (at Library/PackageCache/com.unity.test-framework@1.1.29/UnityEditor.TestRunner/TestLaunchers/TestLauncherBase.cs:27)
UnityEditor.TestTools.TestRunner.TestLauncherBase.ExecuteMethods[T] (NUnit.Framework.Interfaces.ITest tests, NUnit.Framework.Interfaces.ITestFilter testRunnerFilter, UnityEditor.TestTools.TestRunner.AttributeFinderBase attributeFinder, System.String logString, System.Action`1[T] action, System.Nullable`1[T] testTargetPlatform) (at Library/PackageCache/com.unity.test-framework@1.1.29/UnityEditor.TestRunner/TestLaunchers/TestLauncherBase.cs:66)
UnityEditor.EditorApplication:Internal_CallUpdateFunctions()

```

Reply:
Thanks, will add that in. I guess because we're running with Unity open its loaded the Nuget plugin and created the file already for us

Reply:
Added the "LoadNuGetConfigFile" and the new "LoadSettingsFile" calls into the setup phase